test/converters/junitxml/junitxml.go

```diff
@@ -76,22 +76,191 @@ func convertTestReport(report TestReport) testreport.TestReport {
 
 func convertTestSuit(testSuit TestSuite) testreport.TestSuite {
 	convertedTestSuite := testreport.TestSuite{
-		XMLName:  testSuit.XMLName,
-		Name:     testSuit.Name,
-		Tests:    testSuit.Tests,
-		Failures: testSuit.Failures + testSuit.Errors,
-		Skipped:  testSuit.Skipped,
-		Time:     testSuit.Time,
+		XMLName: testSuit.XMLName,
+		Name:    testSuit.Name,
+		Time:    testSuit.Time,
 	}
 
-	for _, testCase := range testSuit.TestCases {
+	tests := 0
+	failures := 0
+	skipped := 0
+
+	flattenedTestCases := flattenGroupedTestCases(testSuit.TestCases)
+	for _, testCase := range flattenedTestCases {
 		convertedTestCase := convertTestCase(testCase)
 		convertedTestSuite.TestCases = append(convertedTestSuite.TestCases, convertedTestCase)
+
+		if convertedTestCase.Failure != nil {
+			failures++
+		}
+		if convertedTestCase.Skipped != nil {
+			skipped++
+		}
+		tests++
 	}
 
+	convertedTestSuite.Tests = tests
+	convertedTestSuite.Failures = failures
+	convertedTestSuite.Skipped = skipped
+
 	return convertedTestSuite
 }
 
+func flattenGroupedTestCases(testCases []TestCase) []TestCase {
+	var flattenedTestCases []TestCase
+	for _, testCase := range testCases {
+		flattenedTestCases = append(flattenedTestCases, testCase)
+
+		if len(testCase.FlakyFailures) == 0 && len(testCase.FlakyErrors) == 0 &&
+			len(testCase.RerunFailures) == 0 && len(testCase.RerunErrors) == 0 {
+			continue
+		}
+
+		flattenedTestCase := TestCase{
+			XMLName:           testCase.XMLName,
+			ConfigurationHash: testCase.ConfigurationHash,
+			Name:              testCase.Name,
+			ClassName:         testCase.ClassName,
+			Time:              0,
+			Failure:           nil,
+			Skipped:           nil,
+			Error:             nil,
+			FlakyFailures:     nil,
+			RerunFailures:     nil,
+		}
+
+		for _, flakyFailure := range testCase.FlakyFailures {
+			flattenedTestCase.Failure = convertFlakyFailureToFailure(flakyFailure)
+			flattenedTestCases = append(flattenedTestCases, flattenedTestCase)
+		}
+
+		for _, flakyError := range testCase.FlakyErrors {
+			flattenedTestCase.Failure = convertFlakyErrorToFailure(flakyError)
+			flattenedTestCases = append(flattenedTestCases, flattenedTestCase)
+		}
+
+		for _, rerunfailure := range testCase.RerunFailures {
+			flattenedTestCase.Failure = convertRerunFailureToFailure(rerunfailure)
+			flattenedTestCases = append(flattenedTestCases, flattenedTestCase)
+		}
+
+		for _, rerunError := range testCase.RerunErrors {
+			flattenedTestCase.Failure = convertRerunErrorToFailure(rerunError)
+			flattenedTestCases = append(flattenedTestCases, flattenedTestCase)
```

Comment:
`convertXXXToFailure` is the same code 4x (insert rant about Go "interfaces").
Could we maybe replace them/delegate to something like `convertToFailure(type, message, systemErr string)`?